src/uagents/network.py

```diff
@@ -26,3 +37,26 @@ def get_faucet() -> FaucetApi:
 
 def get_reg_contract() -> LedgerContract:
     return _contract
+
+
+async def wait_for_tx_to_complete(
+    tx_hash: str,
+    timeout: Optional[timedelta] = None,
+    poll_period: Optional[timedelta] = None,
+) -> TxResponse:
+    if timeout is None:
+        timeout = timedelta(seconds=DEFAULT_QUERY_TIMEOUT_SECS)
+    if poll_period is None:
+        poll_period = timedelta(seconds=DEFAULT_QUERY_INTERVAL_SECS)
```

Comment:
```
timeout = timeout or timedelta(seconds=DEFAULT_QUERY_TIMEOUT_SECS)
poll_period = poll_period or timedelta(seconds=DEFAULT_QUERY_INTERVAL_SECS)
```

Reply:
Good suggestion. This would work in almost all cases, except if someone wanted to set either value to 0 for some reason, so I'll probably leave it the way it is.